src/App/Validator/KohanaValidationEngine.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Ushahidi\App\Validator;
+
+/**
+ * Ushahidi Core Validation Tool
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Core
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html
+ *             GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+use Kohana\Validation\Validation as KohanaValidation;
+use Ushahidi\Core\Tool\ValidationEngine;
+use Illuminate\Contracts\Translation\Translator;
+
+class KohanaValidationEngine extends KohanaValidation implements ValidationEngine
+{
+    /**
+     * @var TranslatorContract
+     */
+    private $translator;
+
+    public function __construct(Translator $translator)
+    {
+        $this->translator = $translator;
+
+        parent::__construct([], $this->getTranslationCallback());
+    }
+
+    public function getTranslationCallback()
+    {
+        return function ($file, $field, $error = null) {
```

Comment:
This is probably the ugliest bit. We pass a callback into the KohanaValidation lib for translating messages, and do some wrangling here to connect that to lumen's translator.